tests/unit/FileExports/CustomerFileExportTest.php

```diff
@@ -14,6 +17,51 @@ public function getFileExportClass(): string
         return CustomerFileExport::class;
     }
 
+    public function dataProviderUserCreateByRoles()
+    {
+        $tests = [];
+        foreach (GOCustomer::VALID_ROLES as $role) {
+            $tests['valid-'.$role] = [$role, $role.'_test'];
+        }
+        foreach (GOCustomer::INVALID_ROLES as $role) {
+            $tests['invalid-'.$role] = [$role, $role.'_test'];
+        }
+
+        return $tests;
+    }
+
+    /**
+     * @dataProvider dataProviderUserCreateByRoles
+     */
+    public function testUserCreateByRoles($role, $role_name): void
+    {
+        $this->initApiConnection();
+        // Setting is_logged_in to true
+        wp_set_current_user(1);
+        StoreKeeperApi::$mockAdapter->withModule(
+            'ShopModule',
+            function (MockInterface $module) {
+                $module->shouldReceive('newShopCustomer')->andReturnUsing(
+                    function ($got) {
+                        $user = $got[0]['relation'];
+                        // Passed firstname as the role in test data
+                        $role = $user['contact_person']['firstname'];
+                        $this->assertNotContains($role, GOCustomer::INVALID_ROLES, $role.' should not be returned');
```

Comment:
this functions is not beeing called then you make the invalid role how does this asset work than?